src/overlays/actors/ovl_Bg_Ydan_Hasi/z_bg_ydan_hasi.c

```diff
@@ -21,23 +32,179 @@ const ActorInit Bg_Ydan_Hasi_InitVars = {
+        } else {
+            // 3 platforms on 2F
+            DynaPolyInfo_Alloc(&D_06005780, &localConst);
+            thisx->draw = NULL;
+            this->actionFunc = BgYdanHasi_SetupThreeBlocks;
+            Actor_SetHeight(thisx, 40.0f);
+        }
+        this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, localConst);
+    }
+    this->timer = 0;
+}
+
+void BgYdanHasi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgYdanHasi* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void BgYdanHasi_UpdateFloatingBlock(BgYdanHasi* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 framesAfterMath;
+    f32 posOffset;
+    WaterBox* waterBox;
+
+    framesAfterMath = sinf((globalCtx->gameplayFrames & 0xFF) * 0.024543693f) * 165.0f;
+    this->dyna.actor.posRot.pos.x =
+        ((Math_Sins(this->dyna.actor.posRot.rot.y) * framesAfterMath) + this->dyna.actor.initPosRot.pos.x);
+    this->dyna.actor.posRot.pos.z =
+        ((Math_Coss(this->dyna.actor.posRot.rot.y) * framesAfterMath) + this->dyna.actor.initPosRot.pos.z);
+    waterBox = &globalCtx->colCtx.stat.colHeader->waterBoxes[1];
+    this->dyna.actor.posRot.pos.y = waterBox->unk_02 + 20.0f;
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->timer == 0) {
+        this->timer = 50;
+    }
+    posOffset = sinf(this->timer * 0.12566371f);
+    this->dyna.actor.posRot.pos.y = (this->dyna.actor.posRot.pos.y + (posOffset + posOffset));
+}
+
+void BgYdanHasi_InitWater(BgYdanHasi* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, this->unk_168)) {
+        this->timer = 600;
+        this->actionFunc = BgYdanHasi_MoveWater;
+    }
+}
+
+WaterBox* BgYdanHasi_MoveWater(BgYdanHasi* this, GlobalContext* globalCtx) {
+    WaterBox* waterBox;
+
+    if (this->timer == 0) {
+        if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y, 1.0f) != 0) {
+            Flags_UnsetSwitch(globalCtx, this->unk_168);
+            this->actionFunc = BgYdanHasi_InitWater;
+        }
+        func_8002F948(&this->dyna.actor, NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+    } else {
+        if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y - 47.0f, 0.5f)) {
+            this->actionFunc = BgYdanHasi_DecWaterTimer;
+        }
+        func_8002F948(&this->dyna.actor, NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+    }
+    waterBox = globalCtx->colCtx.stat.colHeader->waterBoxes;
+    globalCtx->colCtx.stat.colHeader->waterBoxes[1].unk_02 = this->dyna.actor.posRot.pos.y;
+    if (1) {}
+    return waterBox + 0x1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE690.s")
+void BgYdanHasi_DecWaterTimer(BgYdanHasi* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    func_8002F994(&this->dyna.actor, this->timer);
```

Comment:
this function is more than just decrementing a a timer, its also playing a sound if the timer is less than 40.
I dont really have a good name suggestion right now, but dont think it shuold just be named after one of the things that it does. feel free to leave unamed if you cant think of anything